platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/settings/DeclarativeDslProjectSettingsIntegrationSpec.groovy

```diff
@@ -86,6 +87,54 @@ class DeclarativeDslProjectSettingsIntegrationSpec extends AbstractIntegrationSp
         "semantic"         | "x = 1"               | "2:13: unresolved reference 'x'"
     }
 
+    def 'reports illegal order of settings blocks on #order'() {
+        given:
+        file("settings.gradle.dcl") << content
+
+        when:
+        def failure = fails(":projects")
+
+        then:
+        failure.assertHasErrorOutput(errorMessage)
+
+        where:
+        order                                | content                                          | errorMessage
+        'statement before plugin management' | 'rootProject.name = "foo"\npluginManagement { }' | "1:1: illegal content before 'pluginManagement', which can only appear as the first element in the file"
+        'plugins before plugin management'   | 'plugins { }\npluginManagement { }'              | "1:1: illegal content before 'pluginManagement', which can only appear as the first element in the file"
+        'statement before plugins'           | 'rootProject.name = "foo"\nplugins { }'          | "1:1: illegal content before 'plugins', which can only be preceded by 'pluginManagement"
+    }
+
+    def 'reports duplicate #kind blocks in settings'() {
+        given:
+        file("settings.gradle.dcl") << content
+
+        when:
+        def failure = fails(":projects")
+
+        then:
+        failure.assertHasErrorOutput(errorMessage)
+
+        where:
+        kind               | content                                                                             | errorMessage
+        'plugins'          | 'pluginManagement { }\nplugins { }\nrootProject.name = "foo"\nplugins { }'          | ""
+        'pluginManagement' | 'pluginManagement { }\nplugins { }\nrootProject.name = "foo"\npluginManagement { }' | ""
```

Comment:
❓ Is the empty error message intended?

Reply:
Nope, it was an oversight. Fixed in [`73940b1` (#29028)](https://github.com/gradle/gradle/pull/29028/commits/73940b15ca7b4b79ce525af2113726591d61e785).